### PR TITLE
Add onfido

### PR DESCRIPTION
**Pull request by desfero**
To be reviewed after #3802 is merged to master

**Technical notes**
This PR adds Onfido support to our current KYC flow. To run the changes locally you need to use the https://github.com/Neufund/platform-backend/pull/2266 and then talk with me or Dave to get proper config for onfido.

***Technical changes***
- `webpack.dev-external-remote.js` was removed to simplify config.

**TODO**
- [x] Implement success modal 
- [x] Implement not supported message
- [x] Fix onfido modal on mobile device
- [x] Redirect to KYC after clicking continue on the profile page
- [x] Make sure visual regression is correct

**Moved as a separate ticket**
- [x] Implement not supported message for an edge case when user decided to go through mobile verification on desktop (ticket #3816)
- [x] In case KYC is in progress `Continue verification process` should move you to the place where you need to fill missing parts (right now it always moves to the beginning) (ticket #3817)
- [x] Add e2e tests (ticket #3086 )

**Check after KYC ui changes pr is merged to master**
 - [ ] Onfido modal was closed but still not completed so all other providers should be blocked at the moment

### CODE REVIEW

File: packages/web/app/lib/api/kyc/KycApi.ts [hunk 1]
```diff
@@ -310,4 +317,44 @@ export class KycApi {
...
+      body: {
+        referrer,
+      },
+      responseSchema: KycOnfidoUploadRequestSchema,
+    });
+
+    return response.body;
+  }
+
+  public async getOnfidoCheckRequest(): Promise<TKycOnfidoCheckRequest> {
```
Comment (desfero) [inline packages/web/app/lib/api/kyc/KycApi.ts]: TODO: Remove from the api

File: packages/web/@types/onfido-sdk-ui/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+declare module "onfido-sdk-ui" {
```
Comment (desfero) [inline packages/web/@types/onfido-sdk-ui/index.d.ts]: Unfortunately, we don't have typings provided by onfido

File: packages/web/app/components/kyc/KYCprocess.stories.tsx [hunk 3]
```diff
@@ -51,14 +53,27 @@ storiesOf("KYC/process", module).add("personal/start", () => {
...
-  return <KycPersonalDocumentVerificationComponent {...props} />;
-});
+storiesOf("KYC/process", module)
+  .add("personal/documentVerification onfido supported", () => {
+    const props = {
+      onStartIdNow: action("onStartIdNow"),
+      onStartOnfido: action("onStartOnfido"),
+      onManualVerification: action("onManualVerification"),
+      layout: EKycRequestType.INDIVIDUAL,
+      onfidoSdk: { isSupported: () => true } as OnfidoSDK,
```
Comment (desfero) [inline packages/web/app/components/kyc/KYCprocess.stories.tsx]: is there any way to mock something quickly without force casting?

File: packages/web/app/middlewares/redux-injectify.ts [hunk 4]
```diff
@@ -1,64 +0,0 @@
-import { Container } from "inversify";
```
Comment (desfero) [inline packages/web/app/middlewares/redux-injectify.ts]: Not used anymore (probably some leftover from our old way of writing business logic (aka flows)

File: packages/web/app/modules/kyc/instant-id/onfido/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { AppReducer } from "../../../../store";
+import { DeepReadonly } from "../../../../types";
+
+export interface IKycOnfidoState {}
+
+export const onfidoInitialState: IKycOnfidoState = {};
+
+export const onfidoReducer: AppReducer<IKycOnfidoState> = (
```
Comment (desfero) [inline packages/web/app/modules/kyc/instant-id/onfido/reducer.ts]: at the moment it's empty but after we merge with #3812 there will be a followup-up ticket to add error to the view so then we will fill reducer with an implementation

File: packages/web/app/styles/elements/ul-ol.scss [hunk 6]
```diff
@@ -1,17 +0,0 @@
-ul {
```
Comment (desfero) [inline packages/web/app/styles/elements/ul-ol.scss]: this one conflicts with an onfido styles, moved to `Field.tsx` and remove `pure` everywhere it was used.

File: packages/web/app/components/kyc/KYCprocess.stories.tsx (hunk 3 above)
Comment (Moejoe90) [reply packages/web/app/components/kyc/KYCprocess.stories.tsx]: no that i know of. 
We can create util that infers any and then only change one place if we decide to change the behavouir

File: packages/web/app/modules/kyc/instant-id/onfido/sagas.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    });
+
+    return () => {
+      onfidoSDK.removeAllListeners();
+    };
+  });
+
+  try {
+    while (true) {
+      const event: TChannelTypes | END = yield take(channel);
```
Comment (Moejoe90) [inline packages/web/app/modules/kyc/instant-id/onfido/sagas.ts]: do we need to specify a buffer size here? 
To not fall into buffer overflow problems

File: packages/web/app/components/kyc/personal/DocumentVerification.tsx [hunk 8]
```diff
@@ -71,28 +81,64 @@ export const KycPersonalDocumentVerificationComponent: React.FunctionComponent<I
...
     </div>
+
     <div className={styles.description} data-test-id="kyc-panel-description">
       <FormattedHTMLMessage tagName="span" id="kyc.personal.instant-id.description" />
     </div>
     <img className={styles.image} src={idImage} alt="id now" />
     <div className="mb-5 text-center">
       <Button
-        onClick={props.onStartInstantId}
+        onClick={props.onStartIdNow}
```
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/components/kyc/personal/DocumentVerification.tsx]: it's a part of @drakkein pr

File: packages/web/app/lib/dependencies/onfido/onfidoSteps.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+      `'${intl.locale}' locale is used that's not supported by Onfido. Please do proper translation manually.`,
+      new Error("Unsupported locale used to start Onfido sdk"),
+    );
+  }
+
+  return [
+    {
+      type: "welcome",
+      options: {
+        title: intl.formatIntlMessage("kyc.onfido.welcome.title"),
```
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/lib/dependencies/onfido/onfidoSteps.ts]: unfortunately no. it's rendered outside of our app (to be precise it's rendered inside Onfido modal which is a `preact` app)

File: packages/web/app/components/kyc/personal/DocumentVerification.tsx [hunk 10]
```diff
@@ -71,28 +81,64 @@ export const KycPersonalDocumentVerificationComponent: React.FunctionComponent<I
...
+>(
   appConnect<IStateProps, IDispatchProps>({
     dispatchToProps: dispatch => ({
-      onStartInstantId: () => dispatch(actions.kyc.kycStartIndividualIdNow()),
+      onStartIdNow: () => dispatch(actions.kyc.startIdNowRequest()),
+      onStartOnfido: () => dispatch(actions.kyc.startOnfidoRequest()),
       onManualVerification: () => dispatch(actions.routing.goToKYCIndividualUpload()),
     }),
   }),
+  fromRenderProps<TRenderPropsProp, {}, NonNullable<TContainerContext>>(
```
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/components/kyc/personal/DocumentVerification.tsx]: ohh, I was hoping you won't find the duplicates here 😆, fixed 🤖

File: packages/web/app/components/shared/loading-indicator/LoadingIndicator.stories.tsx [hunk 11]
```diff
@@ -6,6 +6,8 @@ import { ELoadingIndicator, LoadingIndicator, LoadingIndicatorContainer } from "
 
 storiesOf("LoadingIndicator", module)
   .add("pulse (default)", () => <LoadingIndicator />)
+  .add("pulse light", () => <LoadingIndicator type={ELoadingIndicator.PULSE} light={true} />)
```
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/components/shared/loading-indicator/LoadingIndicator.stories.tsx]: added to the storybook

File: packages/web/app/modules/kyc/instant-id/id-now/sagas.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+
+function* startIdNow({
+  apiKycService,
+  notificationCenter,
+  logger,
+}: TGlobalDependencies): Iterator<any> {
+  try {
+    const { redirectUrl }: TKycIdNowIdentification = yield apiKycService.startInstantId();
+
+    yield put(actions.routing.openInNewWindow(redirectUrl));
```
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/modules/kyc/instant-id/id-now/sagas.ts]: yeah, after we redirect we just show a message about if the redirect didn't happen please use this link

File: packages/web/app/modules/kyc/sagas.ts [hunk 13]
```diff
@@ -686,4 +667,8 @@ export function* kycSagas(): Iterator<any> {
     actions.kyc.kycStopWatching,
     kycRefreshWidgetSaga,
   );
+
+  // sub-sagas
+  yield fork(kycOnfidoSagas);
```
[the remark this answers is not in the document]
Comment (desfero) [reply packages/web/app/modules/kyc/sagas.ts]: yeah, thats the point, bank transfer flow probably even can be moved outside of `modules/kyc`

**Pull request merged**